### PR TITLE
chore(web-components): [button] move style related api out of base class

### PR DESCRIPTION
**Pull request by eljefe223**
## Previous Behavior
Style specific parts of Button's API were bundled into the class. 

## New Behavior
Style specific parts of Button's API are kept in the `Button` class while functional parts of the API are moved into a new `BaseButton` class. This allows consumers to opt out of style specifics they may not need.

## Related Issue(s)

- Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: packages/web-components/src/button/button.ts [hunk 1]
```diff
@@ -515,6 +424,99 @@ export class Button extends FASTElement {
   }
 }
 
+export class Button extends BaseButton {
```
Comment (radium-v) [inline packages/web-components/src/button/button.ts:427]: This class should have a documentation block to describe its purpose and to differentiate between it and the base class, and to mark it as `@public`.

**Comment by fabricteam** [comment]
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

File: packages/web-components/src/button/button.ts (hunk 1 above)
Comment (eljefe223) [reply packages/web-components/src/button/button.ts:427]: FYI we do not do this in anchor button

Comment (chrisdholt) [reply packages/web-components/src/button/button.ts:427]: We likely should given we're intending to make the API's available.

Comment (eljefe223) [reply packages/web-components/src/button/button.ts:427]: Yup just pointing out other places where we aren't doing this handling those cases now.

File: packages/web-components/src/anchor-button/anchor-button.ts [hunk 2]
```diff
@@ -242,6 +242,12 @@ export class BaseAnchor extends FASTElement {
   }
 }
 
+/**
+ * A Anchor Custom HTML Element.
```
Comment (radium-v) [inline packages/web-components/src/anchor-button/anchor-button.ts:246]:
```suggestion
 * An Anchor Custom HTML Element.
```

**Review by radium-v — APPROVED** [review]

**Review by mlijanto — APPROVED** [review]

**Pull request merged**